spec/requests/api/v1/posts_spec.rb

```diff
@@ -0,0 +1,51 @@
+RSpec.describe Api::V1::PostsController, type: :request do
+let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+
+  describe "GET /api/v1/posts" do
+    posts= Post.all
+
+    before do
+      5.times { FactoryBot.create(:post) }
+    end
+
+    it "returns a collection of posts" do
+      get "/api/v1/posts", headers: headers
+      expect(json_response.count).to eq 5
+    end
+
+    it "returns 200 response" do
+      get "/api/v1/posts", headers: headers
+      expect(response.status).to eq 200
+    end
+    
+    it "has correct keys in the response" do
+      get "/api/v1/posts", headers: headers
+
+      posts.each do |post|
+        expect(json_response[posts.index(post)]).to include('id')
+        expect(json_response[posts.index(post)]).to include('status')
+        expect(json_response[posts.index(post)]).to include('latitude')
+        expect(json_response[posts.index(post)]).to include('longitude')
+      end
+    end
+
+    it "has correct status on response" do
+      get "/api/v1/posts", headers: headers
+     
+      posts.each do |post|
+        expect(post.published?).to be(false)
```

Comment:
These go green because you hardcode the status and category in the factory. There is no need to test that the Factory works properly here

Reply:
Ah yes. Then since we will test the default values for `status`and `category`when we implement `Create`method, we remove these tests now!